### PR TITLE
kernel miniapp (test with laset)

### PR DESCRIPTION
**Pull request by rasolca**
Utilities to easily write miniapp for cuda kernels.

### CODE REVIEW

File: miniapp/include/dlaf/miniapp/kernel_runner.h [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+#include "dlaf/types.h"
+
+#ifdef DLAF_WITH_CUDA
+#include <cuda_runtime.h>
+#endif
+namespace dlaf::miniapp {
+
+template <Backend backend>
+struct KernelRunner;
+
```
[the remark this answers is not in the document]
Comment (rasolca) [reply miniapp/include/dlaf/miniapp/kernel_runner.h:24]: Exactly. The idea is to exclude as much level of complexity as possible. Adding pika means to add runtime and schedulers...

**Review by albestro — APPROVED** [review]
LGTM

File: miniapp/include/dlaf/miniapp/kernel_runner.h [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+struct KernelRunner<Backend::MC> {
+  KernelRunner(SizeType count, SizeType nthreads) noexcept : count_(count), nthreads_(nthreads) {
+    threads_.reserve(nthreads_);
+  }
+
+  // @pre kernel_task should accept only one argument of type SizeType.
+  template <class F>
+  double run(F&& kernel_task) noexcept {
+    auto task = [count = count_, &kernel_task](int id, int tot) {
+      for (SizeType i = id * count / tot; i < (id + 1) * count / tot; ++i) {
```
[the remark this answers is not in the document]
Comment (rasolca) [reply miniapp/include/dlaf/miniapp/kernel_runner.h]: Is it ok now?

[the remark this answers is not in the document]
Comment (msimberg) [reply miniapp/include/dlaf/miniapp/kernel_runner.h]: Yes, thanks!

**Review by msimberg — APPROVED** [review]

**Pull request merged**